tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Web/Common/MatterProvision.cs

```diff
@@ -1147,35 +1170,33 @@ private static string TeamMembersPermissionInformation(MatterDetails matterDetai
-            keys.Add(matterSettings.StampedPropertySubAreaOfLaw);
-            keys.Add(matterSettings.StampedPropertyMatterName);
-            keys.Add(matterSettings.StampedPropertyMatterID);
-            keys.Add(matterSettings.StampedPropertyClientName);
-            keys.Add(matterSettings.StampedPropertyClientID);
-            keys.Add(matterSettings.StampedPropertyResponsibleAttorney);
-            keys.Add(matterSettings.StampedPropertyTeamMembers);
-            keys.Add(matterSettings.StampedPropertyIsMatter);
-            keys.Add(matterSettings.StampedPropertyOpenDate);
-            keys.Add(matterSettings.StampedPropertySecureMatter);
-            keys.Add(matterSettings.StampedPropertyBlockedUploadUsers);
-            keys.Add(matterSettings.StampedPropertyMatterDescription);
-            keys.Add(matterSettings.StampedPropertyConflictCheckDate);
-            keys.Add(matterSettings.StampedPropertyConflictCheckBy);
-            keys.Add(matterSettings.StampedPropertyMatterCenterRoles);
-            keys.Add(matterSettings.StampedPropertyMatterCenterPermissions);
-            keys.Add(matterSettings.StampedPropertyMatterCenterUsers);
-            keys.Add(matterSettings.StampedPropertyDefaultContentType);
-            keys.Add(matterSettings.StampedPropertyIsConflictIdentified);
-            keys.Add(matterSettings.StampedPropertyDocumentTemplateCount);
-            keys.Add(matterSettings.StampedPropertyBlockedUsers);
-            keys.Add(matterSettings.StampedPropertyMatterGUID);
-            keys.Add(matterSettings.StampedPropertyMatterCenterUserEmails);
-
-            propertyList.Add(matterSettings.StampedPropertyPracticeGroup, WebUtility.HtmlEncode(matterDetails.PracticeGroup));
-            propertyList.Add(matterSettings.StampedPropertyAreaOfLaw, WebUtility.HtmlEncode(matterDetails.AreaOfLaw));
-            propertyList.Add(matterSettings.StampedPropertySubAreaOfLaw, WebUtility.HtmlEncode(matterDetails.SubareaOfLaw));
+            //Get all the matter stamped properties from the appsettings.json file
+            var matterStampedProperties = configuration.GetSection("Matter").GetChildren();
+            foreach (var key in matterStampedProperties)
+            {
+                //Assuming that all the keys for the matter property bag keys will start with "StampedProperty"
+                if (key.Key.ToString().ToLower().StartsWith("stampedproperty"))
+                {
+                    keys.Add(key.Key);
+                }
+            }
+            /*
+             * All the managed columns need to be read from the appsettings.json file. In old implementation
+             * all the managed columns are hardcoded and that hardcoding has been removed, by reading the
+             * column names from appsettings.json file
+             */
+            //Get the number of levels from Taxonomy Settings
+            int levels = int.Parse(configuration.GetSection("Taxonomy")["Levels"].ToString());

```

Comment:
Also adding the managed columns to the property bag should be generic as the hierarchy can be dynamic. Today it is 3 and it was hardcoded and tomorrow it may change. This is the same generic logic that has been followed